File: server/models/store.go

package models

type Store struct {
	ID      uint   `gorm:"primaryKey" json:"id"`
	Name    string `gorm:"type:varchar(255)" json:"name"`
	Code    string `gorm:"type:varchar(255)" json:"code"`
	Address string `gorm:"type:varchar(255)" json:"address"`
}

File: store_app/lib/app/data/model/store.dart
class Store {
  final int id;
  final String name;
  final String code;
  final String address;

  Store({required this.id, required this.name, required this.code, required this.address});

  factory Store.fromJson(Map<String, dynamic> json) {
    return Store(
      id: json['id'],
      name: json['name'],
      code: json['code'],
      address: json['address'],
    );
  }

  Map<String, dynamic> toJson() {
    return {'id': id, 'name': name, 'code': code, 'address': address};
  }
}
